source/slang/slang-ir-legalize-types.cpp

```diff
@@ -343,8 +343,20 @@ struct LegalCallBuilder
                 // result type of the function, so we know that
                 // the legalization funciton/call will use a `void`
                 // result type.
-                //
-                _emitCall(m_context->builder->getVoidType());
+
+                // If our call is in global scope (initializer) we should
+                // hoist the call into the start of every entry-point 
+                if (!m_context->builder->getInsertLoc().getInst()->getParent()
```

Comment:
I understand what this is trying to achieve here, but I think this is likely not the right way.

Usually if you have a global variable and you want to initialize it with some code (e.g. a call), you create a `IRGlobalVar`, which is a `IRGlobalValueWithCode` that can have a body just like a `IRFunc`. Inside the body you create a basic block with instructions + a `return` instruct that returns the value of the init expr.

Here instead of inserting a call into global scope, we should be creating that IRGlobalVar, and rely on follow up passes to insert the logic in the body into entry points.